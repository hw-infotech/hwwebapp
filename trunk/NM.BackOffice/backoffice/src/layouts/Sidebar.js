import React, { useState } from "react";
import { Nav } from "react-bootstrap";
import { useLocation, useNavigate } from "react-router";
import { Navigation } from "react-minimal-side-navigation";
import Icon from "awesome-react-icons";

const Sidebar = () => {

    const history = useNavigate();
    const location = useLocation();
    const [isSidebarOpen, setIsSidebarOpen] = useState(true);

    return (
        <div >
            <Navigation
                // you can use your own router's api to get pathname
                activeItemId="/management/members"
                onSelect={({ itemId }) => {
                    // maybe push to the route
                }}
                items={[
                    {
                        title: 'Dashboard',
                        itemId: '/dashboard',
                        // you can use your own custom Icon component as well
                        // icon is optional
                        elemBefore: () => <Icon name="inbox" />,
                    },
                    {
                        title: 'Management',
                        itemId: '/management',
                        elemBefore: () => <Icon name="users" />,
                        subNav: [
                            {
                                title: 'Projects',
                                itemId: '/management/projects',
                            },
                            {
                                title: 'Members',
                                itemId: '/management/members',
                            },
                        ],
                    },
                    {
                        title: 'Another Item',
                        itemId: '/another',
                        subNav: [
                            {
                                title: 'Teams',
                                itemId: '/management/teams',
                            },
                        ],
                    },
                ]}
            />
        </div>
    )
}

export default Sidebar
