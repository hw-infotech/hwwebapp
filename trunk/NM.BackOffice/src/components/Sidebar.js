import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { useNavigate } from "react-router";
import { NavLink } from "react-bootstrap";
import { RiGalleryLine } from "react-icons/ri";
import { BsBuilding } from "react-icons/bs";
import { BiNews } from "react-icons/bi";
import { RiHome2Line } from "react-icons/ri";
import { GoQuestion } from "react-icons/go";

const Sidebar = ({ sidebarShow }) => {
  const [parentId, setParentId] = useState(
    JSON.parse(localStorage.getItem("activeId"))?.split("_")[0]
  );

  const classname = "activeMenu";
  let classes = "reverse";
  const [show, setshow] = useState(false);
  const navigtion = useNavigate();

  useEffect(() => {
    setParentId(JSON.parse(localStorage.getItem("activeId")).split("_")[0]);
  }, [JSON.parse(localStorage.getItem("activeId"))]);

  const setLocalStorage = (id) => {
    localStorage.setItem("activeId", JSON.stringify(id));
    setParentId(JSON.parse(localStorage.getItem("activeId")).split("_")[0]);
  };

  const handleAnchorClick = (e) => {
    e.preventDefault();
    setLocalStorage(e.currentTarget.getAttribute("id"));
  };

  return (
    <div
      className="sidebar"
      style={
        sidebarShow
          ? {
              transform: "translate3d(0%, 0px,0px)",
              transition: "all 0.4s ease-in-out",
              marginLeft: "0px",
            }
          : {
              transform: "translate3d(-100%, 0px, 0px)",
              transition: "all 0.4s ease-in-out",
              marginLeft: "-250px",
            }
      }
    >
      <div
        className="sidebar_header"
        onClick={() => {
          navigtion("/");
        }}
      >
        <img src="./assets/images/NM-ICON.png" className="Sidebar-logo" />
        <div>
          <span className="fs_13 header_text">
            <b>Back</b>Office
          </span>
        </div>
      </div>
      <div className="align-items-sm-start px-3 pt-2">
        <ul className="nav nav-pills sidebar-ul align-items-center" id="menu">
          <li className={`nav-item sidebar_hover`}>
            <a
              className={`nav-link sidebar_hover align-middle px-0 text-color-white gap-icon-text  ${
                localStorage.getItem("className") == "Dashboard" && classname
              }`}
              id="dashboard"
              data-bs-toggle="collapsed"
              onClick={(e) => {
                localStorage.setItem("className", "Dashboard");
                setLocalStorage(" ");
                return navigtion("/dashboard");
              }}
            >
              <RiHome2Line size={20} />
              <span className="ms-1 d-none d-sm-inline sidebar_li_text  ">
                Dashboard
              </span>
            </a>
          </li>
          <li className="sidebar_list">
            <a
              href="#submenu2"
              data-bs-toggle="collapse"
              className={
                parentId == "submenu2"
                  ? "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text"
                  : "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
              }
              onClick={(e) => {
                localStorage.setItem("className", "success");
                // setLocalStorage("submenu2_1");
                // return navigtion("/success-stories");
              }}
            >
              <RiGalleryLine size={20} />
              {/*<img src="./assets/images/car.png" style={{background:"white"}} width={25} height={25}/>*/}
              <span className="ms-1 d-none d-sm-inline sidebar_li_text ">
                Sliders/Carousels
              </span>
              <span className="dropdownarrowicon">
                <i className="fs-6 bi-caret-down"></i>
              </span>
            </a>
            <ul
              className={
                parentId == "submenu2"
                  ? "collapse nav flex-column show"
                  : "collapse nav flex-column "
              }
              id="submenu2"
              data-bs-parent="#menu"
            >
              <li className={`w-100 sidebar_list`}>
                <a
                  id="submenu2_1"
                  className={`nav-link text-color-white gap-icon-text sidebar_focus ${
                    localStorage.getItem("className") == "success" && classname
                  }`}
                  onClick={(e) => {
                    handleAnchorClick(e);
                    localStorage.setItem("className", "success");
                    navigtion("/success-stories");
                  }}
                >
                  <span className="navbar-submenu">Success Stories</span>
                </a>
              </li>
            </ul>
          </li>
          <li className="sidebar_list">
            <a
              onClick={(e) => {
                localStorage.setItem("className", "subscirber");
                // setLocalStorage("submenu8_1");
                // return navigtion("/subscribe-unsubscribe");
              }}
              href="#submenu8"
              data-bs-toggle="collapse"
              className={
                parentId == "submenu8"
                  ? "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text"
                  : "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
              }
            >
              <BiNews size={20} />
              <span className="ms-1 d-none d-sm-inline">Newsletter</span>
              <span className="dropdownarrowicon">
                <i className="fs-6 bi-caret-down "></i>
              </span>
            </a>
            <ul
              className={
                parentId == "submenu8"
                  ? "collapse nav flex-column show"
                  : "collapse nav flex-column "
              }
              id="submenu8"
              data-bs-parent="#menu"
            >
              <li className="w-100 sidebar_inline">
                <a
                  id="submenu8_1"
                  className={`nav-link  text-color-white gap-icon-text sidebar_focus ${
                    localStorage.getItem("className") == "subscirber" &&
                    classname
                  }`}
                  onClick={(e) => {
                    localStorage.setItem("className", "subscirber");
                    handleAnchorClick(e);
                    navigtion("/subscribe-unsubscribe");
                  }}
                >
                  <span className="d-none d-sm-inline navbar-submenu ">
                    Subscribe-Unsubscribe
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li className="sidebar_list">
            <a
              href="#submenu9"
              data-bs-toggle="collapse"
              className={
                parentId == "submenu9"
                  ? "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text"
                  : "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
              }
              onClick={(e) => {
                localStorage.setItem("className", "allenquiry");
                // setLocalStorage("submenu9_1");
                // return navigtion("/all-Enquiry");
              }}
            >
              <GoQuestion size={20} />
              <span className="ms-1 d-none d-sm-inline ">Enquiry</span>
              <span className="dropdownarrowicon">
                <i className="fs-6 bi-caret-down "></i>
              </span>
            </a>
            <ul
              className={
                parentId == "submenu9"
                  ? "collapse nav flex-column show"
                  : "collapse nav flex-column "
              }
              id="submenu9"
              data-bs-parent="#menu"
            >
              <li className="w-100 ">
                <a
                  id="submenu9_1"
                  className={`nav-link  text-color-white gap-icon-text sidebar_focus ${
                    localStorage.getItem("className") == "allenquiry" &&
                    classname
                  }`}
                  onClick={(e) => {
                    handleAnchorClick(e);
                    localStorage.setItem("className", "allenquiry");
                    navigtion("/all-Enquiry");
                  }}
                >
                  <span className="d-none d-sm-inline navbar-submenu ">
                    All Enquiry
                  </span>
                </a>
              </li>
            </ul>
          </li>
          <li className="sidebar_list">
            <a
              href="#submenu10"
              data-bs-toggle="collapse"
              className={
                parentId == "submenu10"
                  ? "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text"
                  : "nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
              }
              onClick={(e) => {
                localStorage.setItem("className", "joblist");
                // setLocalStorage("submenu10_1");
                // return navigtion("/all-jobs");
              }}
            >
              <BsBuilding size={20} />
              <span className=" d-none d-sm-inline">Job Management</span>
              <span className="dropdownarrowicon">
                <i className="fs-6 bi-caret-down "></i>
              </span>
            </a>
            <ul
              className={
                parentId == "submenu10"
                  ? "collapse nav flex-column show"
                  : "collapse nav flex-column "
              }
              id="submenu10"
              data-bs-parent="#menu"
            >
              <li className="sidebar_list">
                <a
                  id="submenu10_1"
                  className={`nav-link  text-color-white gap-icon-text sidebar_focus ${
                    localStorage.getItem("className") == "joblist" && classname
                  }`}
                  onClick={(e) => {
                    handleAnchorClick(e);
                    localStorage.setItem("className", "joblist");
                    navigtion("/all-jobs");
                  }}
                >
                  <span className="d-none d-sm-inline navbar-submenu">Job</span>
                </a>
              </li>
              <li>
                <a
                  id="submenu10_2"
                  className={`nav-link  text-color-white gap-icon-text sidebar_focus ${
                    localStorage.getItem("className") == "jobsub" && classname
                  }`}
                  onClick={(e) => {
                    handleAnchorClick(e);
                    localStorage.setItem("className", "jobsub");
                    navigtion("/job-subscribe-unsubscribe");
                  }}
                >
                  <span className="d-none d-sm-inline navbar-submenu">
                    Subscribe-Unsubscribe
                  </span>
                </a>
              </li>
            </ul>
          </li>
        </ul>
        <hr />
      </div>
    </div>
  );
};
export default Sidebar;
