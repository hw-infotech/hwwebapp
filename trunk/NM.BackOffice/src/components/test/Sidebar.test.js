import {
  fireEvent,
  getAllByRole,
  getByTestId,
  getByText,
  render,
  screen,
  within,
} from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Sidebar from "../Sidebar";

describe("Sidebar testing", () => {
  test("Sidebar snapshot testing", () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Sidebar />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
