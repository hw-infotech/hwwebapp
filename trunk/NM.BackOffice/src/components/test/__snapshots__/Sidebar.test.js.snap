// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar testing Sidebar snapshot testing 1`] = `
<div
  className="sidebar"
  style={
    Object {
      "marginLeft": "-250px",
      "transform": "translate3d(-100%, 0px, 0px)",
      "transition": "all 0.4s ease-in-out",
    }
  }
>
  <div
    className="sidebar_header"
  >
    <img
      className="Sidebar-logo"
      src="./assets/images/NM-ICON.png"
    />
    <div>
      <span
        className="fs_13 header_text"
      >
        <b>
          Back
        </b>
        Office
      </span>
    </div>
    <button
      className="move-sidebar positioning-header btn"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        color="black"
        fill="currentColor"
        height={30}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "black",
          }
        }
        viewBox="0 0 12 16"
        width={30}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.41 9H.59C0 9 0 8.59 0 8c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zm0-4H.59C0 5 0 4.59 0 4c0-.59 0-1 .59-1H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1h.01zM.59 11H11.4c.59 0 .59.41.59 1 0 .59 0 1-.59 1H.59C0 13 0 12.59 0 12c0-.59 0-1 .59-1z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
  <div
    className="align-items-sm-start px-3 pt-2"
  >
    <ul
      className="nav nav-pills sidebar-ul align-items-center"
      id="menu"
    >
      <li
        className="nav-item sidebar_hover"
      >
        <a
          className="nav-link sidebar_hover align-middle px-0 text-color-white gap-icon-text  false"
          data-bs-toggle="collapsed"
          id="dashboard"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 21H5a1 1 0 0 1-1-1v-9H1l10.327-9.388a1 1 0 0 1 1.346 0L23 11h-3v9a1 1 0 0 1-1 1zM6 19h12V9.157l-6-5.454-6 5.454V19z"
              />
            </g>
          </svg>
          <span
            className="ms-1  d-sm-inline sidebar_li_text  "
          >
            Dashboard
          </span>
        </a>
      </li>
      <li
        className="sidebar_list"
      >
        <a
          className="nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
          data-bs-toggle="collapse"
          href="#submenu2"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 13c-1.678 0-3.249.46-4.593 1.259A14.984 14.984 0 0 1 18.147 19H20v-6zm-3.996 6C14.044 14.302 9.408 11 4 11v8h12.004zM4 9c3.83 0 7.323 1.435 9.974 3.796A10.949 10.949 0 0 1 20 11V3h1.008c.548 0 .992.445.992.993v16.014a1 1 0 0 1-.992.993H2.992A.993.993 0 0 1 2 20.007V3.993A1 1 0 0 1 2.992 3H6V1h2v4H4v4zm14-8v4h-8V3h6V1h2zm-1.5 9a1.5 1.5 0 1 1 0-3 1.5 1.5 0 0 1 0 3z"
              />
            </g>
          </svg>
          <span
            className="ms-1 d-sm-inline sidebar_li_text  ripple-surface "
            tabindex="1"
          >
            Sliders/Carousels
          </span>
          <span
            className="dropdownarrowicon"
          >
            <i
              className="fs-6 bi-caret-down"
            />
          </span>
        </a>
        <ul
          className="collapse nav flex-column "
          data-bs-parent="#menu"
          id="submenu2"
        >
          <li
            className="w-100 sidebar_list"
          >
            <a
              className="nav-link text-color-white gap-icon-text sidebar_focus false"
              id="submenu2_1"
              onClick={[Function]}
            >
              <span
                className="navbar-submenu"
              >
                Success Stories
              </span>
            </a>
          </li>
        </ul>
      </li>
      <li
        className="sidebar_list"
      >
        <a
          className="nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
          data-bs-toggle="collapse"
          href="#submenu8"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.875 3H4.125C2.953 3 2 3.897 2 5v14c0 1.103.953 2 2.125 2h15.75C21.047 21 22 20.103 22 19V5c0-1.103-.953-2-2.125-2zm0 16H4.125c-.057 0-.096-.016-.113-.016-.007 0-.011.002-.012.008L3.988 5.046c.007-.01.052-.046.137-.046h15.75c.079.001.122.028.125.008l.012 13.946c-.007.01-.052.046-.137.046z"
            />
            <path
              d="M6 7h6v6H6zm7 8H6v2h12v-2h-4zm1-4h4v2h-4zm0-4h4v2h-4z"
            />
          </svg>
          <span
            className="ms-1 d-sm-inline  sidebar_li_text"
          >
            Newsletter
          </span>
          <span
            className="dropdownarrowicon"
          >
            <i
              className="fs-6 bi-caret-down "
            />
          </span>
        </a>
        <ul
          className="collapse nav flex-column "
          data-bs-parent="#menu"
          id="submenu8"
        >
          <li
            className="w-100 sidebar_inline"
          >
            <a
              className="nav-link  text-color-white gap-icon-text sidebar_focus false"
              id="submenu8_1"
              onClick={[Function]}
            >
              <span
                className=" navbar-submenu d-sm-inline "
              >
                Subscribe-Unsubscribe
              </span>
            </a>
          </li>
        </ul>
      </li>
      <li
        className="sidebar_list"
      >
        <a
          className="nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
          data-bs-toggle="collapse"
          href="#submenu9"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 14 16"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 10h2v2H6v-2zm4-3.5C10 8.64 8 9 8 9H6c0-.55.45-1 1-1h.5c.28 0 .5-.22.5-.5v-1c0-.28-.22-.5-.5-.5h-1c-.28 0-.5.22-.5.5V7H4c0-1.5 1.5-3 3-3s3 1 3 2.5zM7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7z"
              fillRule="evenodd"
            />
          </svg>
          <span
            className="ms-1  d-sm-inline "
          >
            Enquiry
          </span>
          <span
            className="dropdownarrowicon"
          >
            <i
              className="fs-6 bi-caret-down "
            />
          </span>
        </a>
        <ul
          className="collapse nav flex-column "
          data-bs-parent="#menu"
          id="submenu9"
        >
          <li
            className="w-100 "
          >
            <a
              className="nav-link  text-color-white gap-icon-text sidebar_focus false"
              id="submenu9_1"
              onClick={[Function]}
            >
              <span
                className=" navbar-submenu "
              >
                All Enquiry
              </span>
            </a>
          </li>
        </ul>
      </li>
      <li
        className="sidebar_list"
      >
        <a
          className="nav-link px-0 sidebar_focus align-middle text-color-white gap-icon-text collapsed"
          data-bs-toggle="collapse"
          href="#submenu10"
          onClick={[Function]}
        >
          <svg
            fill="currentColor"
            height={20}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width={20}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.763.075A.5.5 0 0 1 15 .5v15a.5.5 0 0 1-.5.5h-3a.5.5 0 0 1-.5-.5V14h-1v1.5a.5.5 0 0 1-.5.5h-9a.5.5 0 0 1-.5-.5V10a.5.5 0 0 1 .342-.474L6 7.64V4.5a.5.5 0 0 1 .276-.447l8-4a.5.5 0 0 1 .487.022zM6 8.694 1 10.36V15h5V8.694zM7 15h2v-1.5a.5.5 0 0 1 .5-.5h2a.5.5 0 0 1 .5.5V15h2V1.309l-7 3.5V15z"
              fillRule="evenodd"
            />
            <path
              d="M2 11h1v1H2v-1zm2 0h1v1H4v-1zm-2 2h1v1H2v-1zm2 0h1v1H4v-1zm4-4h1v1H8V9zm2 0h1v1h-1V9zm-2 2h1v1H8v-1zm2 0h1v1h-1v-1zm2-2h1v1h-1V9zm0 2h1v1h-1v-1zM8 7h1v1H8V7zm2 0h1v1h-1V7zm2 0h1v1h-1V7zM8 5h1v1H8V5zm2 0h1v1h-1V5zm2 0h1v1h-1V5zm0-2h1v1h-1V3z"
            />
          </svg>
          <span
            className="  d-sm-inline"
          >
            Job Management
          </span>
          <span
            className="dropdownarrowicon"
          >
            <i
              className="fs-6 bi-caret-down "
            />
          </span>
        </a>
        <ul
          className="collapse nav flex-column "
          data-bs-parent="#menu"
          id="submenu10"
        >
          <li
            className="sidebar_list"
          >
            <a
              className="nav-link  text-color-white gap-icon-text sidebar_focus false"
              id="submenu10_1"
              onClick={[Function]}
            >
              <span
                className=" d-sm-inline navbar-submenu"
              >
                Job
              </span>
            </a>
          </li>
          <li>
            <a
              className="nav-link  text-color-white gap-icon-text sidebar_focus false"
              id="submenu10_2"
              onClick={[Function]}
            >
              <span
                className=" d-sm-inline navbar-submenu"
              >
                Subscribe-Unsubscribe
              </span>
            </a>
          </li>
        </ul>
      </li>
    </ul>
    <hr />
  </div>
</div>
`;
